hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,164 @@
+The default highlighting of tokens according to the syntactic type is then altered according to the semantic token type and the color theme selected by the user (editor preferences).
+
+In Langium, the `SemanticTokenProvider` service is responsible for assigning language-dependent semantic types to tokens.
+Therefore, we customize the default semantic token provider like this:
+
+```ts
+import { AbstractSemanticTokenProvider, AstNode, SemanticTokenAcceptor } from "langium";
+import { isPerson } from "./generated/ast.js";
+import { SemanticTokenTypes } from 'vscode-languageserver';
+
+export class HelloWorldSemanticTokenProvider extends AbstractSemanticTokenProvider {
+    protected override highlightElement(node: AstNode, acceptor: SemanticTokenAcceptor): void | "prune" | undefined {
+        if (isPerson(node)) {
+            acceptor({
+                node,
+                property: 'name',
+                type: SemanticTokenTypes.class
+            });
+        }
+    }
+}
+```
+
+For all persons (`isPerson(...)` in line 7), we explicitly specify the semantic type for the token of their `'name'` property.
+Here, we use `SemanticTokenTypes.class` as semantic type.
+For your case, select a predefined type which fits your domain best.
+Since the name is used as cross-reference by greetings, a similar check and assignment of a semantic token type needs to be done for the `person` property of `Greeting` as well.
+
+After creating the semantic token provider, you need to register the `HelloWorldSemanticTokenProvider` in `hello-world-module.ts` in the following way:
+
+```ts
+export const HelloWorldModule: Module<HelloWorldServices, PartialLangiumServices & HelloWorldAddedServices> = {
+    lsp: {
+        SemanticTokenProvider: (services) => new HelloWorldSemanticTokenProvider(services)
+    },
+    validation: {
+        HelloWorldValidator: () => new HelloWorldValidator()
+    }
+};
+```
+
+Now rebuild and restart your application and test the improvements of the second step:
+
+![screenshot with fixed semantic token](fixed-2-token.png)
+
+The `HelloWorldSemanticTokenProvider` works, and you might see a different highlighting XOR you might not see any difference, e.g. "Hello" is still blue here. This leads us to the third step.
+
+
+## Step 3: Ensure that your editor styles the chosen semantic token type
+
+The __third step__ is to ensure, that your editor supports the assigned semantic tokens:
+Depending on your editor and the currently selected color theme, the semantic token type selected in `HelloWorldSemanticTokenProvider` might not be supported or didn't got a different color in the color theme.
+The easiest way to detect such problems is to change the current color theme and to try some others.
+Note, that VS Code allows to switch off semantic highlighting for all themes with the setting `editor.semanticHighlighting.enabled`.
```

Comment:
"in front of _that_ there is no comma" 🎶

```suggestion
Note that VS Code allows to switch off semantic highlighting for all themes with the setting `editor.semanticHighlighting.enabled`.
```

Reply:
done